docs/src/pages/ThemingSpecification.mdx

```diff
@@ -0,0 +1,297 @@
+| Simple definition of tokens per parent component                                                                  | YES                                         | The parent component can define tokens for all children                                                                                        | NO                                       | Each component can define it's own token, which can be problematic when we want to override something on parent's lavel |
+| Adding new design term in the parent requires re-composing all slots components                                   | NO                                          | We can just override styles in the slot's styles inside the parents                                                                            | YES                                      | We need to re-compose if we want new design terms in the slots components                                               |
+| Requires knowledge of the internals of another component, which creates coupling                                  | PARTIALLY                                   | The parent can define styles only for the root (no other slots), but it needs to know some details for the root's styles of the slot component | NO                                       | All styles are defined in the composed component                                                                        |
+| Easy to swap a slot into another component which might want all its own styles.                                   | NO                                          |                                                                                                                                                | YES                                      |
+
+##### Variants
+
+Variants are just mechanism for allowing users do add or remove some design terms from their components.
+
+In order to enable this, the compose mechanism should provide to the users adding/removing some of the design terms. This can be easily done, by allowing the user to define mapping from props to design terms when they define the composed components.
+
+```
+export interface FluentButton extends ButtonProps {
+  primary?: boolean
+  circular?: boolean
+}
+
+// these code snippet is just illustrating a possible
+// mechanism of how the design tokens can be created
+// based on the props
+const FluentButton = compose(Button, {
+  displayName: 'FluentButton',
+  mapPropsToTerms: props => ({
+    primary: props.primary,
+    circular: props.circular,
+    hasIcon: !!props.icon,
+  }),
+})
+```
+
+##### Animations
+
+Animation should be defined as part of the theme, but they should not be all loaded if they are not used. Child themes should be able to customize them.
+
+```
+// TBD
+```
+
+##### Icons
+
+Icons should be defined as part of the theme, but they should not loaded if not used. Child themes should be able to customize them.
+
```

Comment:
Icons is a huge chuck of TeamsTheme. Are there ways to reduce the content? (I believe it's all the SVG content.) 

In Fabric we started with a HUGE icon set. I ended up separating the icons into chunks: one PLT1 chunk, and a number of other chunks which could be loaded on demand or deferred. This meant that some icons would be defined asynchronously. 

This would be something to consider here. Having icons be part of the theme would require a re-render of the entire tree in order to add more icons. (Maybe!)

Reply:
Yeah, I feel like we need to do a bit more investigation before deciding what is the best, that's why I didn't feel this section... I will add this as an info in the section, thanks for sharing!

Reply:
Icons already created a lot of complexity and I am not sure that they are really part of the theme.